### PR TITLE
Menu Copy and Paste

### PR DESCRIPTION
**Pull request by sarinali**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Bugzilla bug ID

**ID:** 1902755
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1902755

## Type of change

Please delete options that are not relevant.

- [x] New Test
- [x] New POM

# How does this resolve / make progress on that bug?

Done.

# Screenshots / Explanations

N/A

# Comments / Concerns

N/A

### CODE REVIEW

File: tests/menus/test_copy_paste_actions.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    with driver.context(driver.CONTEXT_CHROME):
+        context_menu.get_context_item("context-menu-paste").click()
+        login_fill.hide_popup("contentAreaContextMenu")
+
+    # final assertion
+    assert password_field.get_attribute("value") != random_text
+
+
+def test_text_area_copy_paste(driver: Firefox):
+    # initialize obbjects
```
Comment (ben-c-at-moz) [inline tests/menus/test_copy_paste_actions.py:46]: spelling

File: tests/menus/test_copy_paste_actions.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    # copy the text
+    text_area_fill.triple_click("street-address-textarea")
+    context_menu.context_click_element(text_area)
+    with driver.context(driver.CONTEXT_CHROME):
+        context_menu.get_context_item("context-menu-copy").click()
+        text_area_fill.hide_popup("contentAreaContextMenu")
+
+    # delete all the text and paste
+    text_area.send_keys(Keys.BACK_SPACE)
```
Comment (ben-c-at-moz) [inline tests/menus/test_copy_paste_actions.py:65]: Should probably check that the text area is cleared.

File: tests/menus/test_copy_paste_actions.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    google_search.triple_click("search-bar-textarea")
+
+    # context click
+    context_menu.context_click_element(search_bar)
+    with driver.context(driver.CONTEXT_CHROME):
+        context_menu.get_context_item("context-menu-copy").click()
+        google_search.hide_popup("contentAreaContextMenu")
+
+    # delete the current text
+    search_bar.send_keys(Keys.BACK_SPACE)
```
Comment (ben-c-at-moz) [inline tests/menus/test_copy_paste_actions.py:97]: Should probably check that the search bar is cleared.

**Review by ben-c-at-moz — CHANGES_REQUESTED** [review]
A few small things, which is great for a test of this size. Good job!

File: tests/menus/test_copy_paste_actions.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    # delete all text
+    password_field.send_keys(Keys.BACK_SPACE)
+    context_menu.context_click_element(password_field)
+    with driver.context(driver.CONTEXT_CHROME):
+        context_menu.get_context_item("context-menu-paste").click()
+        login_fill.hide_popup("contentAreaContextMenu")
+
+    # final assertion
+    assert password_field.get_attribute("value") != random_text
```
[the remark this answers is not in the document]
Comment (sarinali) [reply tests/menus/test_copy_paste_actions.py:42]: you would have to ensure that the clipboard is clear before testing, this is OS specific which is more work and this has about the same effect

**Review by ben-c-at-moz — APPROVED** [review]
let's gooooooo

**Pull request merged**